lib/internal/util.js

```diff
@@ -60,6 +60,9 @@ exports._deprecate = function(fn, msg) {
   var warned = false;
   function deprecated() {
     warned = exports.printDeprecationMessage(msg, warned, deprecated);
+    if (new.target) {
+      return new (Function.prototype.bind.apply(fn, arguments));
```

Comment:
I think a test should be added with a subclass as well to make confirm the functionality